### ISSUE TITLE
Random hanging

### ISSUE THREAD

**Issue by PhantomGamers** [CrendKing/avisynth_filter#41]
### Environment
- Filter version: v0.9.4
- AviSynth+ version: 3.7.0
- OS: Windows 10 x64
- GPU: NVIDIA GTX 1080 (driver 461.81)
- CPU: Intel Core i9 10850k (10c20t)
- Video player: MPC-BE x64 1.5.6
- Splitter filter and video decoder: MPC
- Video renderer: madVR
- Video format: NV12
- AviSynth script: SVP
- Do you use other DirectShow filters? Specify name and version: No

### Describe the bug
After watching a video for a period of time (it doesn't seem very consistent, sometimes it happens within 5 minutes, sometimes within 15, ~15 minutes tends to be the longest), with madVR the video frame freezes but the audio continues to play, and pausing the video and replaying it causes it to start playing video again but no longer at the interpolated framerate (not positive what is happening here, is it SVP's script? avisynth filter? avisynth+? How can I find out?) When using EVR the entire player just crashes. I have zero problems using SVP with just ffdshow (well, as far as crashes/hanging goes).

I'd love to figure out how to get this working as the one major change over ffdshow that helps so much for the experience is that with ffdshow, the video will freeze for a bit after AR changes and SVP's black bar detection kicks in. With avisynth filter it transitions nearly seamlessly.

I thought maybe this could have been related to #24 but I've tried messing with the CPU and GPU queue sizes and it seems to happen regardless of what they are set to.

I should also mention that I tried it with the AvySynth+ version that gets installed by SVP (3.5 I think?), and it also occurred there, I manually updated it to 3.7.0 seeing that it had improvements for multithreading and thought maybe it would help, but alas it did not.

**Checklist**
- Try a different **video player** and check if the issue persist. For example, if your main video player is MPC-HC, try MPC-BE.
Findings: I've tested and this happens in both MPC-HC and MPC-BE.

- Try a different **video file** and check if the issue persist. Try another file with different format, dimension and frame rate.
Findings: I've been having this consistently with everything I watch so I don't believe the file format is related.

[avsf.zip (log file)](https://github.com/CrendKing/avisynth_filter/files/6138279/avsf.zip)

**Comment by CrendKing** [comment]
Without a consistent repro, I can't pinpoint the root cause. But I found a pattern in your log file that shares with https://github.com/CrendKing/avisynth_filter/issues/40. I already fixed that problem in https://github.com/CrendKing/avisynth_filter/commit/fe73211d792b8b4d827f0c35d31f77f4bee7b29b. Can you try [AviSynthFilter.zip](https://github.com/CrendKing/avisynth_filter/files/6144822/AviSynthFilter.zip)?

**Comment by PhantomGamers** [comment]
> Without a consistent repro, I can't pinpoint the root cause.

Yeah I know, it's really annoying on my end as well, but it's never consistent. It'll hang, and if I restart the video file from the beginning it'll hang at a completely different part...

And it still happens with the version you've posted here.

Something else I've noticed, using ffdshow instead of avisynth filter I sometimes get frames with green artifacts (it happens so fast that I'm unable to get a screenshot of it), so I'm wondering if perhaps there's a problem somewhere else in the pipeline that ffdshow just gracefully recovers from whereas avisynth filter does not?

Here are some more logs of freezes in case it helps anything
[avsf.zip](https://github.com/CrendKing/avisynth_filter/files/6152534/avsf.zip)

**Comment by CrendKing** [comment]
I found a concurrency issue for #40. I'm not sure if it's the same issue, but it doesn't hurt trying: [AviSynthFilter.zip](https://github.com/CrendKing/avisynth_filter/files/6156028/AviSynthFilter.zip).

**Comment by PhantomGamers** [comment]
I think I can confirm that this is some issue between avisynth filter and SVP's black bar detection & removal.

I watched through a whole show and it didn't freeze once, and what I've noticed is that there were zero A/R changes throughout, whereas the other 2 shows I was watching had quite a few.

I didn't consider trying this before because as I said, I wasn't getting crashing with ffdshow, but after making the link I went back to the most crash-prone parts of the other shows and didn't get any crashing with the black bar detection disabled.

I've attached another avsf log file as well as the event log from SVP (no errors in there).
Not sure if I should also post this on the SVP forums / through SVP or if continuing this discussion here is appropriate? @chainikdn 
[(avsf log) avsf.zip](https://github.com/CrendKing/avisynth_filter/files/6161422/avsf.zip)
[(svp log) active.zip](https://github.com/CrendKing/avisynth_filter/files/6161423/active.zip)

**Comment by CrendKing** [comment]
Interesting. Can you post screenshot of your SVP settings (especially the black bar stuff) so I can replicate the environment?

**Comment by PhantomGamers** [comment]
![image](https://user-images.githubusercontent.com/844685/111582929-f415a900-8791-11eb-88f5-d65f663390f3.png)
![image](https://user-images.githubusercontent.com/844685/111582941-fbd54d80-8791-11eb-8700-536e99e92abc.png)

Everything should be default except for the frame conversion being set to "To screen"

Perhaps this is being a little overcautious, but in case some DMCA bot picks up the names, imdb:tt9140560 and imdb;tt1898069 were the key offenders in triggering this (for the former, it seemed especially bad in 4k but I also had some issues with 1080p sources so that might be irrelevant I'm not positive)

**Comment by CrendKing** [comment]
Sorry. I never used the black bar feature. I see that some part of S03E05 has no black bars and then later there are. Does SVP automatically detects these bars on realtime and adjust, or you have to manually trigger something to detect? If automatic, does it hang whenever that change happens?

**Comment by PhantomGamers** [comment]
It automatically detects the changes in realtime and adjusts, but it usually doesn't happen immediately on a switch, usually it will switch and actually continue playing for a few seconds (sometimes minutes, even, which is another reason I had trouble tying the two together). The part from about 36:30 to 41:00 is the worst of it, seemingly guaranteed to hang (though not always at the same exact part).

**Comment by CrendKing** [comment]
I played 4 times with different setup and no hang at all. But more importantly, why would anyone want to use that black bar feature during this scene? Isn't it extremely annoying to see the video size constantly changing and stutter?

**Comment by chainikdn** [comment]
> why would anyone want to use that black bar feature during this scene? Isn't it extremely annoying to see the video size constantly changing and stutter?

off-topic here, but this feature is either on or off... I don't know how it could ignore this scene only
though it's set to "one time only" by default meaning it will cut black bars one time at the beginning of playback

**Comment by CrendKing** [comment]
> though it's set to "one time only" by default meaning it will cut black bars one time at the beginning of playback

That makes sense. I can't find any toggle to force it one time only.

**Comment by chainikdn** [comment]
there's a popup menu on the "Black bars detection" button ;)

**Comment by CrendKing** [comment]
Oh. What a sneaky menu! It is default to Tracking mode though:
![Clipboard01](https://user-images.githubusercontent.com/975235/111600687-e4e12c00-878e-11eb-9a3c-9bb3d4b93d0f.jpg)

@PhantomGamers So changing to "One time only" mode may be a workaround for you now until we gather more info on repro?

**Comment by chainikdn** [comment]
> It is default to Tracking mode though

mmm... could be :D I'm still not sure which one should be the default and it used to change several times in the past

**Comment by PhantomGamers** [comment]
> But more importantly, why would anyone want to use that black bar feature during this scene? Isn't it extremely annoying to see the video size constantly changing and stutter?

Yes, during this scene of course I wouldn't want the black bar detection to be on, but it was hanging in basically every episode of the show even where it would only transition once on a scene change.

>  though it's set to "one time only" by default meaning it will cut black bars one time at the beginning of playback

For me it was set to tracking mode by default, hmm

**Comment by CrendKing** [comment]
> When you tried to repro, did you use a 4k source? It could potentially be related if you didn't

No. I could try it.

**Comment by chainikdn** [comment]
I can't find s3 in 4k o_O

**Comment by PhantomGamers** [comment]
> I can't find s3 in 4k o_O

I couldn't find every episode in 4k but S03E05 was one of the ones that I could.

If either of you have discord I can send it to you directly, my discord is the same as my Github #9208

**Comment by CrendKing** [comment]
Does it have 1,633,697,175 bytes? I played several times with those tons of AR changes and had no hang at all. I did untick P010 and P016 in AVSF settings to force LAV to convert it to 8-bit YV12 format.

**Comment by PhantomGamers** [comment]
> Does it have 1,633,697,175 bytes?

The 4k sources I'm using are around the 12-13GB range, with that specific episode being 14,590,133,052 bytes

How can you change the AVSF settings while using it with SVP? When I try to untick those settings and save it says "AviSynth script file is currently managed by remote control. Your change If any is saved but not used."

I don't think this show is available in HDR though, maybe it is but I don't have an HDR monitor and the freezing occurs even with HDR related settings disabled in SVP.

If neither of you can reproduce this issue, it could be a hardware issue on my end. I've just upgraded my cpu+mobo (the issue was occurring prior, but I didn't report it until after I upgraded because my old cpu was faulty and I thought it might have been that). And I might have gotten a RTX 3080 for close to MSRP (we'll see if it ships lol), so if/when I get that I'll be able to rule out the GPU as being problematic. Then the only thing remaining would be my RAM, which I have tested extensively and I do not believe it to be problematic.

Also I just clean installed Windows when I got the new cpu/mobo, out of an abundance of caution, so if it's not a hardware issue and neither of you can reproduce it, I'm not sure what kind of software configuration would cause such a thing... hmm

**Comment by CrendKing** [comment]
> How can you change the AVSF settings while using it with SVP

The "Input Formats" below is orthogonal to the avs script and can be set independently.

**Comment by Disaer** [comment]
Hello.
I also got frequent hangs while watching 4K HDR with the "Decrease to screen" SVP option enabled, especially on seek or audio track change. Is this bug reproduced or additional samples are needed?

**Comment by PhantomGamers** [comment]
> I also got frequent crashes while watching 4K HDR with the "Decrease to screen" SVP option enabled

Yes! It seems this alone is responsible for it. I can't seem to reproduce the crashing/hanging after switching it to "Keep normal size."

**Comment by CrendKing** [comment]
Try [AviSynthFilter.zip](https://github.com/CrendKing/avisynth_filter/files/6195410/AviSynthFilter.zip). I don't guarantee a fix though. Let me know the result.

**Comment by PhantomGamers** [comment]
I can still trigger the freezes in "Decrease to screen size" mode with that version, I don't have a reliable repro without that setting.

I think the hangs that happen in "Decrease to screen size" mode and the hangs that happen sporadically are separate issues as I was getting hanging last night while watching the latest episode of imdb:tt6470478 which doesn't have any AR changes.

I should have saved the avsf log of that but I didn't, I will try and get one when I have a chance.

**Comment by CrendKing** [comment]
Is every video that hanged 4K? Could it be that too much memory were used so that additional memory can't be allocated?

Also, [MPC-HC](https://github.com/clsid2/mpc-hc) has a feature that when some error happens, it displays a message box with some info. It also has crash reporter (but I have no access to it. clsid should have). So maybe you can try it and somehow get some info?

Also, if you are developer and have Visual Studio, when it hangs, you can attach VS and pause to get stack trace.

**Comment by PhantomGamers** [comment]
Nope, the sporadic crashes happen with 1080p videos as well! It just seems to happen more often at 4k.

When I have a chance I will try and get a stack trace of the hang.

**Comment by CrendKing** [comment]
Does [v0.9.5 Beta](https://github.com/CrendKing/avisynth_filter/releases/tag/v0.9.5) fix the problem?

**Comment by PhantomGamers** [comment]
Sadly not.

I've made a dump file of the mpc-be process after a hang, not sure if this will help you.
https://pixeldra.in/u/wH9L2srk

[and here's another avsf log just in case](https://github.com/CrendKing/avisynth_filter/files/6226246/avsf.log)

**Comment by PhantomGamers** [comment]
> If you told me you are capable of doing crash dump yourself, we could have made progress long time ago.

This was my first time having to manually create a crash dump, as opposed to one being generated automatically, so I wasn't sure that I was capable of it myself until you suggested attaching the vs debugger lol

> Hope you have a good fiber network for another 600MB upload ;-)

I have a gigabit connection but sadly I cannot upload to pixeldrain at those speeds!
If I had a way of communicating with you directly I could send it to you directly through my hfs server but I would not want to publish my IP address here.

[Here is the dump file with the debug build](https://pixeldra.in/u/sd44ti1a) (it's twice as large because I did it with the reproducible segment which is from a 4k source, the one I posted yesterday isn't so reproducible (it'd take like 30 minutes to hang).

[And here's another avsf.log](https://github.com/CrendKing/avisynth_filter/files/6230539/avsf.log)

Also I suppose this probably isn't directly related to the hanging, since like I said it also happens in sources that have no a/r changes, but this is what gets thrown when a/r changes happen:
> Exception thrown at 0x00007FFF8A1CD759 in mpc-be64.exe: Microsoft C++ exception: IScriptEnvironment::NotFound at memory location 0x000000230757C3E0.
Exception thrown at 0x00007FFF8A1CD759 in mpc-be64.exe: Microsoft C++ exception: IScriptEnvironment::NotFound at memory location 0x000000230757C3E0.
Exception thrown at 0x00007FFF8A1CD759 in mpc-be64.exe: Microsoft C++ exception: ReturnExprException at memory location 0x000000230757D3C0.
Exception thrown at 0x00007FFF8A1CD759 in mpc-be64.exe: Microsoft C++ exception: [rethrow] at memory location 0x0000000000000000.
Exception thrown at 0x00007FFF8A1CD759 in mpc-be64.exe: Microsoft C++ exception: ReturnExprException at memory location 0x000000230757D3C0.

**Comment by CrendKing** [comment]
Try [avsf.zip](https://github.com/CrendKing/avisynth_filter/files/6232063/avsf.zip)

**Comment by PhantomGamers** [comment]
With that version, it hangs immediately with any source.

> Exception thrown at 0x00007FFF8A1CD759 in mpc-be64.exe: Microsoft C++ exception: ReturnExprException at memory location 0x00000070B944D3D0.
Exception thrown at 0x00007FFF8A1CD759 in mpc-be64.exe: Microsoft C++ exception: [rethrow] at memory location 0x0000000000000000.
Exception thrown at 0x00007FFF8A1CD759 in mpc-be64.exe: Microsoft C++ exception: ReturnExprException at memory location 0x00000070B944D3D0.

[avsf.log](https://github.com/CrendKing/avisynth_filter/files/6234472/avsf.log)

[And the dump file](https://mega.nz/file/xAQwSZrS#psM4zurfLFKSmd41LLdXzG6vhAlAJeg-OhP5JokIQUQ)

**Comment by CrendKing** [comment]
My bad. I'll give you another one.

BTW, those exceptions are bogus. They are seemingly harmless. So forget about them.

**Comment by CrendKing** [comment]
Lots of changes in this one. I thought one line was enough. Turns out it causes dead lock thus hang. I need to revamp the whole synchronization mechanisms in avsf.

Expect crash, but mainly let me know if it still hangs.

[AviSynthFilter.zip](https://github.com/CrendKing/avisynth_filter/files/6243223/AviSynthFilter.zip)

**Comment by PhantomGamers** [comment]
Yes, it still hangs with that version. Additionally, I managed to get my hands on a 3090 so I can confirm it's not the old GPU that was responsible for the hanging. The only thing that remains in my system is the RAM, which I've tested extensively so I don't think it's related to that.

[Dump file](https://mega.nz/file/pFZFjQSA#rCqONmR_rC4Pd6U6Wz6GrwM6k_o2LgGYHlnSZCpJyGU)
[avsf.log](https://github.com/CrendKing/avisynth_filter/files/6246788/avsf.log)

**Comment by CrendKing** [comment]
The hang happened at exactly the same location, while now a lock is held in the new version. So I have no idea anymore why it hangs when we deliver frames to madVR for rendering.

Does it hang if you use EVR? If yes, please upload a dump.

**Comment by PhantomGamers** [comment]
It does hang in EVR, which is progress from 0.94 which would just straight up cause the player to crash in EVR!

[Dump file](https://mega.nz/file/YZQGGTAA#_TFfOLn_LUPDTBdXW_UfZOmxUID5AhWzO8CDr5UWqkc)
[avsf.log](https://github.com/CrendKing/avisynth_filter/files/6247193/avsf.log)

**Comment by CrendKing** [comment]
The EVR hang is caused by AviSynth+, which I have no control of.

And I'm confused with your madVR avsf.log. It doesn't seem in sync with that uploaded dump.

**Comment by PhantomGamers** [comment]
> The EVR hang is caused by AviSynth+, which I have no control of.  

I can open an issue over there then. Is there anything specific you noticed that I should pass along?

> And I'm confused with your madVR avsf.log. It doesn't seem in sync with that uploaded dump.  

My bad, not sure what happened. Here is another dump/avsf.log with madVR.

[Dump](https://mega.nz/file/9c4nQCCJ#dz44TdFM9__UFeubuBnTiWlULCQHOXVmQu3maSICHCE)
[avsf.log](https://github.com/CrendKing/avisynth_filter/files/6249605/avsf.log)

**Comment by CrendKing** [comment]
Try [avisynth_filter_64.zip](https://github.com/CrendKing/avisynth_filter/files/6252500/avisynth_filter_64.zip)

**Comment by PhantomGamers** [comment]
Still hangs
[Dump](https://mega.nz/file/BAAXzAgI#Rwx5szU-PgDXseQxxuhhOBsDJQZKKWLk4X6iPGG2x8o)
[avsf.log](https://github.com/CrendKing/avisynth_filter/files/6252612/avsf.log)

**Comment by CrendKing** [comment]
What about? Also if madVR hangs, please try EVR.
[avisynth_filter_64.zip](https://github.com/CrendKing/avisynth_filter/files/6253993/avisynth_filter_64.zip)

**Comment by PhantomGamers** [comment]
Both madvr and evr still hang.

I figured out why the avsf log and dump file didn't match up before, it's because sometimes when mpc crashes the process doesn't actually terminate and so I was attached to the wrong process. Or at least I think that is what happened. These should hopefully be good.

[evr dump](https://mega.nz/file/dZJTUCqQ#lcN1kTJM_hbeUiZ36RqWmGxKZhJp9AWZ6BxPy8agbIk)
[evr avsf.log](https://github.com/CrendKing/avisynth_filter/files/6255682/avsf.log)

[madvr dump](https://mega.nz/file/ZUQjWA4b#Y-iDX-lFa8sFWNNjnXN6_Jx-p1gmauxeOSuMv6K5EBs)
[madvr avsf.log](https://github.com/CrendKing/avisynth_filter/files/6255697/avsf.log)

**Comment by CrendKing** [comment]
I'm sorry the dumps don't seem to match. Could you please upload again with this file?
[avisynth_filter_64.zip](https://github.com/CrendKing/avisynth_filter/files/6256357/avisynth_filter_64.zip)

But from the log I don't think my fix actually matters if the hang happens there. Or it could be that there were multiple sources of hangs, and we fixed some while others still exist (the earlier hangs would have "hidden" later hangs), because I no longer see a hang location in your recent uploads that I once saw in your earlier upload.

**Comment by CrendKing** [comment]
Another supposed improvement: [avisynth_filter_64.zip](https://github.com/CrendKing/avisynth_filter/files/6263378/avisynth_filter_64.zip)

**Comment by CrendKing** [comment]
FYI, I [fixed issue 24](https://github.com/CrendKing/avisynth_filter/issues/24#issuecomment-814657073_), which could help eliminating some hang. You might want to try.

**Comment by PhantomGamers** [comment]
I will test soon! I've been busy with midterms so haven't had a chance to test your changes.

**Comment by ssssssbbb** [comment]
> Lots of changes in this one. I thought one line was enough. Turns out it causes dead lock thus hang. I need to revamp the whole synchronization mechanisms in avsf.
> 
> Expect crash, but mainly let me know if it still hangs.
> 
> [AviSynthFilter.zip](https://github.com/CrendKing/avisynth_filter/files/6243223/AviSynthFilter.zip)

With this version, I can prevent hanging by setting the "rendering device" to "CPU only" in SVP, no matter how much the queue size is.

Hanging comes much faster in every version after this. The last version in issue 24 is even worse, it hangs every time  I right click in PotPlayer.

Please tell me how to produce dump or log.  But 我传不了mega，能QQ或度盘吗？

**Comment by Disaer** [comment]
I've sent a sample via PM on the Doom9 forum. it crashes when setting the "Decrease to screen" SVP option.

**Comment by PhantomGamers** [comment]
> FYI, I [fixed issue 24](https://github.com/CrendKing/avisynth_filter/issues/24#issuecomment-814657073_), which could help eliminating some hang. You might want to try.

With this version I get this error immediately on loading the video, and it pops up again immediately after if clicking continue.
> ASSERT Failed
> height == abs(videoFormat.bmi.biHeight)
> At line 213 of
> D:\Development\C++\DirectShow\avisynth_filter\avisynth_filter\src\format.cpp
[dump](https://mega.nz/file/lAJhCSRT#dUdpugWnrtgfTdpPuhk2IBZUl4VE8NUTVr-_DhkbIqM)
[avsf.log](https://github.com/CrendKing/avisynth_filter/files/6281778/avsf.log)

@ssssssbbb You can generate an avsf log file by following https://github.com/CrendKing/avisynth_filter/wiki/How-to-enable-logging
As for the dump file, you have to use Visual Studio and go to the Debug tab at the top and then attach to process and select your video player process. Not sure if it matters but what I do is open the video player without any video, attach the debugger, and then load the video and reproduce the hang. After the hang, I press the pause button in the Visual Studio toolbar at the top and then go to the Debug menu and select "Save dump as" to save the dump file, and then I zip it up (because these files are usually very large) and upload it to mega. You can use any file host that can support the file size (google drive, mega, pixeldrain, etc)

**Comment by CrendKing** [comment]
Sorry. The issue 24 version has a bug. That could cause some crashes you guys experience and the ASSERT fail error.

Hopefully this should have fixed the problem. Let me know. [avisynth_filter_64.zip](https://github.com/CrendKing/avisynth_filter/files/6283363/avisynth_filter_64.zip)

**Comment by PhantomGamers** [comment]
That version seems to have fixed something as it no longer hangs in the specific part it was hanging in prior, and I can't even get it to hang 100% of the time throughout the entire section but sometimes it does. Here is one instance:

[dump](https://mega.nz/file/RdhU3AiL#At4x6mW6E6Iokha_N-vv7_o58i5zXTk5HlEAHLBZKtY)
[avsf.log](https://github.com/CrendKing/avisynth_filter/files/6283792/avsf.log)

**Comment by CrendKing** [comment]
Thanks. It looks like madVR just hang for no apparent reason. Does EVR hang too?

**Comment by ssssssbbb** [comment]
@PhantomGamers Thanks, seems it's too complicated for me to create a dump file. I know nothing about Visual Studio.
[0409.log](https://github.com/CrendKing/avisynth_filter/files/6285662/0409.log)

@CrendKing 
Here is the log, it didn't even start playing,  PotPlayer freezed with OSD showing "Opening". It's a little different from the former versions, this time it didn't turn "no response" , after I closed PotPlayer through the taskbar icon, the UI was gone but the process kept active till I killed it.

Tried EVR and got same result.

**Comment by PhantomGamers** [comment]
> Thanks. It looks like madVR just hang for no apparent reason. Does EVR hang too?

I played through the section a few times in EVR and could not seem to get it to hang

**Comment by CrendKing** [comment]
@ssssssbbb PotPlayer works fine here, so it could be a separate issue. Please create a new issue and follow the instruction to provide some info.

@PhantomGamers Good to hear. The last dump you uploaded basically shows that avsf delivered a video frame to madVR. madVR internally called some functions and eventually waited for some locks. Since madVR is not open sourced, we don't know what's happening. You could do two things:

1) Try to capture another dump and upload. We can see if we could spot a different cause of hang.
2) Send that dump to madVR somehow (they used to have an issue tracker). They might be able to figure it out.

**Comment by PhantomGamers** [comment]
I opened up a bug report with madVR here: http://bugs.madshi.net/view.php?id=667

So we'll see what they say. I'll post another dump next time I trigger a hang.

**Comment by CrendKing** [comment]
I managed to find a way to repro the hang you experienced. It is because unlike EVR, when being queried for media type change, madVR calls to source filter (most likely LAV Splitter) for its connection media type. This call tries to acquire LAV filter's lock, and consequently form a chain of dead lock.

Specifically,

Main thread: holds LAV `m_csFilter`, blocked on LAV `m_csReceive`
Streaming thread: holds LAV `m_csReceive`, blocked on avsf `m_csReceive`
avsf worker thread: holds avsf `m_csReceive`, blocked on LAV `m_csFilter` <- this is where the problem occurred

In contrast, EVR doesn't seem to make that call, thus no dead lock formed.

Now, I'm not saying this is a bug in madVR. They could do it for good reason. On my hand, I could change my logic to do output format change differently. It may take some time to try different approaches. So I might release a new version with this as a known issue.

Again, if you happen to find a hang with different cause, please follow up.

**Comment by ssssssbbb** [comment]
I found some detials,  it seems to be related to the video encode format? 2 HVC1 encoded clips hang here and another hang randomly. H264 and AVC1 clips hang randomly just like before. All random hanging become much more frequent when switching among play/pause, minimize/fullscreen, forward/backward and haste/slow, I used hotkey.

Maybe it's still the same source with this issue, since it didn't stuck with HVC1 clips in that version I mentioned before.

Should I upload the magnent link of the trouble clips here?

**Comment by PhantomGamers** [comment]
madshi requested that I use the debug symbols for avisynth filter in order to make a freeze report more useful to them.

Is the codebase on github up to date? Can I just build it rn and use the debug symbols and ax from my build?
If not can you post the debug symbols for the version you want me to be testing on?

**Comment by CrendKing** [comment]
I believe I fixed all hangs in this version, including the one madVR one I just mentioned. Please try.

[avisynth_filter_64.zip](https://github.com/CrendKing/avisynth_filter/files/6291443/avisynth_filter_64.zip)

@ssssssbbb If the clip is copyrighted, please do not upload here. You could send me private message in [Doom9](https://forum.doom9.org/member.php?u=231402). Or this email: x14ld8sqc@relay.firefox.com. First try the new version though.

**Comment by ssssssbbb** [comment]
H264 is OK now. HVC1 is still the same.

Email sent.

**Comment by CrendKing** [comment]
@ssssssbbb Fixed the bug. Try the link above again.

**Comment by ssssssbbb** [comment]
@CrendKing Quick test is OK now. I will do more tests. Thanks!!!

**Comment by CrendKing** [comment]
[avisynth_filter_64.zip](https://github.com/CrendKing/avisynth_filter/files/6293061/avisynth_filter_64.zip) might fix it. "Might" because I don't know the cause of that.

**Comment by ssssssbbb** [comment]
> [avisynth_filter_64.zip](https://github.com/CrendKing/avisynth_filter/files/6293061/avisynth_filter_64.zip) might fix it. "Might" because I don't know the cause of that unless you can give me more info (in a separate issue please).
It's worse, hangs much easier.

New issue created.

**Comment by PhantomGamers** [comment]
> I believe I fixed all hangs in this version, including the one madVR one I just mentioned. Please try.

I managed to catch a hang with this version as well:
[dump](https://mega.nz/file/MQxFBaDb#vcLGou2aG87uGFeRk6VYlmVYZCvq_GUU6n9hjrq0PaA)
[avsf.log](https://github.com/CrendKing/avisynth_filter/files/6307768/avsf.log)

**Comment by CrendKing** [comment]
This is the other madVR hang that I can't do anything about. You might want to report to madshi.

I will upload a whole package with debug symbols, so that you can give them more info, when I release the next version.

For now, if no other hang happens anymore, and EVR works without problem, please close the issue for me.

**Comment by CrendKing** [comment]
You can find the debug package here: https://github.com/CrendKing/avisynth_filter/releases/tag/v0.10.0
